### ISSUE TITLE
[GLPI 9.5.7] PHP errors in Processing ticket/change/problem tab

### ISSUE THREAD

**Issue by shoothub** [glpi-project/glpi#10467]
### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Version

9.5.7

### Bug description

Hello,
after update to 9.5.7 I see PHP error messages after creating or openting existing ticket/change/problem which has only one record (no tasks, no follow-ups or solution) in corresponding section: Processing ticket, Processing change, Processing problem.

Below requester user name in Actions historical there are 2 PHP error messages:
`Notice: Undefined variable: userdata in /usr/share/glpi/inc/commonitilobject.class.php on line 7550`
`Notice: Trying to access array offset on value of type null in /usr/share/glpi/inc/commonitilobject.class.php on line 7550`

![glpi-9 5 7-new-item](https://user-images.githubusercontent.com/28008456/151565937-1eca1c1d-3c48-4d3d-9980-10834ff348cf.png)

If I add new task, follow-up or solution error disappear.

![glpi-9 5 7-new-item-followup](https://user-images.githubusercontent.com/28008456/151566281-49a89b52-a9bc-4f83-996f-f04a570b374f.png)

Bug is also present with updated database from 9.5.6 or empty database (new installation)
I use OracleLinux 8.5 with latest updates and PHP 7.4.27 from remi repo and httpd 2.4.37

### Relevant log output
/var/log/glpi/php-errors.log
```
[2022-01-28 16:01:54] glpiphplog.NOTICE:   *** PHP Notice (8): Undefined variable: userdata in /usr/share/glpi/inc/commonitilobject.class.php at line 7550
  Backtrace :
  inc/ticket.class.php:776                           CommonITILObject->showTimeline()
  inc/commonglpi.class.php:637                       Ticket::displayTabContentForItem()
  ajax/common.tabs.php:106                           CommonGLPI::displayStandardTab()
  
[2022-01-28 16:01:54] glpiphplog.NOTICE:   *** PHP Notice (8): Trying to access array offset on value of type null in /usr/share/glpi/inc/commonitilobject.class.php at line 7550
  Backtrace :
  inc/ticket.class.php:776                           CommonITILObject->showTimeline()
  inc/commonglpi.class.php:637                       Ticket::displayTabContentForItem()
  ajax/common.tabs.php:106                           CommonGLPI::displayStandardTab()
```

### Steps To reproduce

1. Create new ticket/problem/change or 
2. Open existing ticket/problem/change without follow-up, task or solution
3. Open tab: **Processing ticket/Processing problem/Processing change**
4. There is a PHP error messages between requester username and i icon in **Actions historical**:
* Notice: Undefined variable: userdata in /usr/share/glpi/inc/commonitilobject.class.php on line 7550
* Notice: Trying to access array offset on value of type null in /usr/share/glpi/inc/commonitilobject.class.php on line 7550
5. Error disappear after creating follow-up, task or solution.

### Your GLPI setup information

```
GLPI 9.5.7 ( => /usr/share/glpi)
Installation mode: TARBALL
Current language:en_GB

Server
 
Operating system: Linux glpi.domain.com 5.4.17-2136.302.7.2.1.el8uek.x86_64 #2 SMP Tue Jan 18 12:11:34 PST 2022 x86_64
PHP 7.4.27 fpm-fcgi (Core, PDO, Phar, Reflection, SPL, SimpleXML, Zend OPcache, apcu, bcmath, bz2, calendar, cgi-fcgi, ctype,
	curl, date, dom, exif, fileinfo, filter, ftp, gd, gettext, hash, iconv, intl, json, ldap, libxml, mbstring, mysql, mysqli,
	mysqlnd, openssl, pcre, pdo_mysql, pdo_sqlite, session, sockets, sodium, sqlite3, standard, tokenizer, xml, xmlreader, xmlrpc,
	xmlwriter, xsl, zip, zlib)
Setup: max_execution_time="600" memory_limit="128M" post_max_size="16M" safe_mode="" session.save_handler="files"
	upload_max_filesize="20M" 
Software: Apache ()
	Mozilla/5.0 (X11; Fedora; Linux x86_64; rv:96.0) Gecko/20100101 Firefox/96.0
Server Software: MariaDB Server
	Server Version: 10.3.28-MariaDB
	Server SQL Mode: 
	Parameters: glpi_user@localhost/glpi_db2
	Host info: Localhost via UNIX socket
	
PHP version is at least 7.2.0 - Perfect!PHP version is at least 7.2.0 - Perfect!
Sessions support is available - Perfect!Sessions support is available - Perfect!
Allocated memory > 64 Mio - Perfect!Allocated memory > 64 Mio - Perfect!
mysqli extension is installedmysqli extension is installed
ctype extension is installedctype extension is installed
fileinfo extension is installedfileinfo extension is installed
json extension is installedjson extension is installed
mbstring extension is installedmbstring extension is installed
iconv extension is installediconv extension is installed
zlib extension is installedzlib extension is installed
curl extension is installedcurl extension is installed
gd extension is installedgd extension is installed
simplexml extension is installedsimplexml extension is installed
intl extension is installedintl extension is installed
ldap extension is installedldap extension is installed
apcu extension is installedapcu extension is installed
Zend OPcache extension is installedZend OPcache extension is installed
xmlrpc extension is installedxmlrpc extension is installed
exif extension is installedexif extension is installed
zip extension is installedzip extension is installed
bz2 extension is installedbz2 extension is installed
sodium extension is installedsodium extension is installed
Database version seems correct (10.3.28) - Perfect!Database version seems correct (10.3.28) - Perfect!

The log file has been created successfully.The log file has been created successfully.
Write access to /var/lib/glpi/files/_cache has been validated.Write access to /var/lib/glpi/files/_cache has been validated.
Write access to /etc/glpi has been validated.Write access to /etc/glpi has been validated.
Write access to /var/lib/glpi/files/_cron has been validated.Write access to /var/lib/glpi/files/_cron has been validated.
Write access to /var/lib/glpi/files has been validated.Write access to /var/lib/glpi/files has been validated.
Write access to /var/lib/glpi/files/_dumps has been validated.Write access to /var/lib/glpi/files/_dumps has been validated.
Write access to /var/lib/glpi/files/_graphs has been validated.Write access to /var/lib/glpi/files/_graphs has been validated.
Write access to /var/lib/glpi/files/_lock has been validated.Write access to /var/lib/glpi/files/_lock has been validated.
Write access to /var/lib/glpi/files/_pictures has been validated.Write access to /var/lib/glpi/files/_pictures has been validated.
Write access to /var/lib/glpi/files/_plugins has been validated.Write access to /var/lib/glpi/files/_plugins has been validated.
Write access to /var/lib/glpi/files/_rss has been validated.Write access to /var/lib/glpi/files/_rss has been validated.
Write access to /var/lib/glpi/files/_sessions has been validated.Write access to /var/lib/glpi/files/_sessions has been validated.
Write access to /var/lib/glpi/files/_tmp has been validated.Write access to /var/lib/glpi/files/_tmp has been validated.
Write access to /var/lib/glpi/files/_uploads has been validated.Write access to /var/lib/glpi/files/_uploads has been validated.


Warning:  mkdir(): Permission denied in /usr/share/glpi/inc/toolbox.class.php on line 1483

The directory could not be created in /usr/share/glpi/marketplace.The directory could not be created in /usr/share/glpi/marketplace.
SELinux boolean httpd_can_network_connect_db is off, some features may require this to be on.SELinux boolean httpd_can_network_connect_db is off, some features may require this to be on.

GLPI constants

 
GLPI_ROOT: /usr/share/glpi
GLPI_CONFIG_DIR: /etc/glpi
GLPI_VAR_DIR: /var/lib/glpi/files
GLPI_LOG_DIR: /var/log/glpi
GLPI_SYSTEM_CRON: 1
GLPI_MARKETPLACE_DIR: /usr/share/glpi/marketplace
GLPI_USE_CSRF_CHECK: 1
GLPI_CSRF_EXPIRES: 7200
GLPI_CSRF_MAX_TOKENS: 100
GLPI_USE_IDOR_CHECK: 1
GLPI_IDOR_EXPIRES: 7200
GLPI_ALLOW_IFRAME_IN_RICH_TEXT: 
GLPI_TELEMETRY_URI: https://telemetry.glpi-project.org
GLPI_INSTALL_MODE: TARBALL
GLPI_NETWORK_MAIL: glpi@teclib.com
GLPI_NETWORK_SERVICES: https://services.glpi-network.com
GLPI_MARKETPLACE_PRERELEASES: 
GLPI_MARKETPLACE_ALLOW_OVERRIDE: 1
GLPI_MARKETPLACE_MANUAL_DOWNLOADS: 1
GLPI_USER_AGENT_EXTRA_COMMENTS: 
GLPI_AJAX_DASHBOARD: 1
GLPI_CALDAV_IMPORT_STATE: 0
GLPI_DEMO_MODE: 0
GLPI_FORCE_EMPTY_SQL_MODE: 1
GLPI_DOC_DIR: /var/lib/glpi/files
GLPI_CACHE_DIR: /var/lib/glpi/files/_cache
GLPI_CRON_DIR: /var/lib/glpi/files/_cron
GLPI_DUMP_DIR: /var/lib/glpi/files/_dumps
GLPI_GRAPH_DIR: /var/lib/glpi/files/_graphs
GLPI_LOCAL_I18N_DIR: /var/lib/glpi/files/_locales
GLPI_LOCK_DIR: /var/lib/glpi/files/_lock
GLPI_PICTURE_DIR: /var/lib/glpi/files/_pictures
GLPI_PLUGIN_DOC_DIR: /var/lib/glpi/files/_plugins
GLPI_RSS_DIR: /var/lib/glpi/files/_rss
GLPI_SESSION_DIR: /var/lib/glpi/files/_sessions
GLPI_TMP_DIR: /var/lib/glpi/files/_tmp
GLPI_UPLOAD_DIR: /var/lib/glpi/files/_uploads
GLPI_NETWORK_REGISTRATION_API_URL: https://services.glpi-network.com/api/registration/
GLPI_MARKETPLACE_PLUGINS_API_URI: https://services.glpi-network.com/api/glpi-plugins/
GLPI_I18N_DIR: /usr/share/glpi/locales
GLPI_VERSION: 9.5.7
GLPI_SCHEMA_VERSION: 9.5.7
GLPI_MIN_PHP: 7.2.0
GLPI_YEAR: 2022

Libraries
 
htmlawed/htmlawed version 1.2.5 in (/usr/share/glpi/vendor/htmlawed/htmlawed)
phpmailer/phpmailer version 6.1.6 in (/usr/share/glpi/vendor/phpmailer/phpmailer/src)
simplepie/simplepie version 1.5.6 in (/usr/share/glpi/vendor/simplepie/simplepie/library)
tecnickcom/tcpdf version 6.3.5 in (/usr/share/glpi/vendor/tecnickcom/tcpdf)
michelf/php-markdown in (/usr/share/glpi/vendor/michelf/php-markdown/Michelf)
true/punycode in (/usr/share/glpi/vendor/true/punycode/src)
iamcal/lib_autolink in (/usr/share/glpi/vendor/iamcal/lib_autolink)
sabre/dav in (/usr/share/glpi/vendor/sabre/dav/lib/DAV)
sabre/http in (/usr/share/glpi/vendor/sabre/http/lib)
sabre/uri in (/usr/share/glpi/vendor/sabre/uri/lib)
sabre/vobject in (/usr/share/glpi/vendor/sabre/vobject/lib)
laminas/laminas-cache in (/usr/share/glpi/vendor/laminas/laminas-cache/src)
laminas/laminas-i18n in (/usr/share/glpi/vendor/laminas/laminas-i18n/src)
laminas/laminas-serializer in (/usr/share/glpi/vendor/laminas/laminas-serializer/src)
monolog/monolog in (/usr/share/glpi/vendor/monolog/monolog/src/Monolog)
sebastian/diff in (/usr/share/glpi/vendor/sebastian/diff/src)
elvanto/litemoji in (/usr/share/glpi/vendor/elvanto/litemoji/src)
symfony/console in (/usr/share/glpi/vendor/symfony/console)
scssphp/scssphp in (/usr/share/glpi/vendor/scssphp/scssphp/src)
laminas/laminas-mail in (/usr/share/glpi/vendor/laminas/laminas-mail/src/Protocol)
laminas/laminas-mime in (/usr/share/glpi/vendor/laminas/laminas-mime/src)
rlanvin/php-rrule in (/usr/share/glpi/vendor/rlanvin/php-rrule/src)
blueimp/jquery-file-upload in (/usr/share/glpi/vendor/blueimp/jquery-file-upload/server/php)
ramsey/uuid in (/usr/share/glpi/vendor/ramsey/uuid/src)
psr/log in (/usr/share/glpi/vendor/psr/log/Psr/Log)
psr/simple-cache in (/usr/share/glpi/vendor/psr/simple-cache/src)
mexitek/phpcolors in (/usr/share/glpi/vendor/mexitek/phpcolors/src/Mexitek/PHPColors)
guzzlehttp/guzzle in (/usr/share/glpi/vendor/guzzlehttp/guzzle/src)
guzzlehttp/psr7 in (/usr/share/glpi/vendor/guzzlehttp/psr7/src)
wapmorgan/unified-archive in (/usr/share/glpi/vendor/wapmorgan/unified-archive/src)
paragonie/sodium_compat in (/usr/share/glpi/vendor/paragonie/sodium_compat/src)

SQL replicas

 
Not active

Notifications

Way of sending emails: PHP

Mails receivers

Plugins list
```

**Comment by cedric-anne** [comment]
Hi,

Already fixed by #10461.

Regards

**Comment by shoothub** [comment]
Thanks a lot for quick response, changing line in commonitilobject.class.php fixed the problem.
